src/__test__/graphql-flow.test.js

```diff
@@ -430,6 +437,59 @@ describe('graphql-flow generation', () => {
         });
     });
 
+    it('should generate all types when exportAllObjectTypes is set', () => {
+        const result = rawQueryToFlowTypes(
+            `
+            query SomeQuery {
+                human(id: "Han Solo") {
+                    id
+                    name
+                    homePlanet
+                    hands
+                    alive
+                    friends {
+                        __typename
+                        ... on Human {
+                            hands
+                        }
+                    }
+                }
+            }
+        `,
+            {exportAllObjectTypes: true},
+        );
+
+        expect(result).toMatchInlineSnapshot(`
+            // SomeQueryType.js
+            export type SomeQueryType = {|
+                variables: {||},
+                response: {|
+
+              /** A human character*/
+              human: ?SomeQuery_human
+            |}
+            |};
+            export type SomeQuery_human = {|
+              alive: ?boolean,
+              friends: ?$ReadOnlyArray<?SomeQuery_human_friends>,
```

Comment:
I like that we defined types in other defined types when this option is enabled.